frontend/src/lib/hooks/useFloatingContainerContext.ts

```diff
@@ -11,6 +11,13 @@ import { createContext, RefObject, useContext } from 'react'
 
 export const FloatingContainerContext = createContext<RefObject<HTMLElement> | undefined>(undefined)
 
-export const useFloatingContainerContext = (): RefObject<HTMLElement> | undefined => {
-    return useContext(FloatingContainerContext)
+export const useFloatingContainer = (): HTMLElement | null | undefined => {
+    const ref = useContext(FloatingContainerContext)
+    const [el, setEl] = useState<HTMLElement | null | undefined>(undefined)
+
+    useEffect(() => {
+        setEl(ref?.current)
+    }, [ref])
+
+    return el
```

Comment:
ELI5 why switch to returning the el instead of the ref?

Reply:
Ah, cos we were always just using the el whenever we returned the ref 🙌

Reply:
Yeah, it is also because the component doesn't re-render from a change in `useContext`, we need the `useState` value to be updated